jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/mapper/JPAMapper.java

```diff
@@ -81,23 +87,53 @@ public Long findMapping(Context context, String ownerId) {
         	}
         }
     }
-    
+
+    @Override
+    public Long findLogMapping(Context<?> context, String ownerId) {
+        EntityManagerInfo info = getEntityManager(context);
+        EntityManager em = info.getEntityManager();
+        try {
+            if (context.getContextId() == null) {
+                return null;
+            }
+
+            Query findQuery = em.createNamedQuery("FindContextMapingByAuditContextId", ContextMappingInfo.class)
+                                .setParameter("contextId", context.getContextId().toString())
+                                .setParameter("ownerId", ownerId);
+
+            ContextMappingInfo contextMappingInfo = (ContextMappingInfo) findQuery.getSingleResult();
+            if (contextMappingInfo != null) {
+                return contextMappingInfo.getKsessionId();
```

Comment:
Not sure if for the future we shouldn't directly return just Long from the DB. This could probably improve performance.

Reply:
context mapping info is necessary stuff for computing the kie session, either you get 3 fields or one, I don't think we will get much different. In any case without numbers is difficult to say

Reply:
Might be.